ui/hooks/useAccountTotalCrossChainFiatBalance.ts

```diff
@@ -0,0 +1,119 @@
+import { shallowEqual, useSelector } from 'react-redux';
+import { toChecksumAddress } from 'ethereumjs-util';
+import {
+  getCurrentCurrency,
+  getNetworkConfigurationsByChainId,
+  getCrossChainTokenExchangeRates,
+  getCrossChainMetaMaskCachedBalances,
+} from '../selectors';
+import {
+  getValueFromWeiHex,
+  sumDecimals,
+} from '../../shared/modules/conversion.utils';
+import { getCurrencyRates } from '../ducks/metamask/metamask';
+import { getTokenFiatAmount } from '../helpers/utils/token-util';
+import { TokenWithBalance } from '../components/app/assets/asset-list/asset-list';
+
+type AddressBalances = {
+  [address: string]: number;
+};
+
+export type Balances = {
+  [id: string]: AddressBalances;
+};
+
+export const useAccountTotalCrossChainFiatBalance = (
+  account: { address: string },
+  formattedTokensWithBalancesPerChain: {
+    chainId: string;
+    tokensWithBalances: TokenWithBalance[];
+  }[],
+) => {
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const currencyRates = useSelector(getCurrencyRates);
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const crossChainContractRates = useSelector(
+    getCrossChainTokenExchangeRates,
+    shallowEqual,
+  );
+
+  const crossChainCachedBalances: Balances = useSelector(
+    getCrossChainMetaMaskCachedBalances,
+  );
+
+  // const loading = false; //todo check if loading is still needed
```

Comment:
Is this still needed?

Reply:
Yes I think this can now be removed. Brian suggested that we can remove it when he got his 44 upgrade in

Reply:
I removed this one 🙏